blockly/src/utils/BlocklyCommands.java

```diff
@@ -186,29 +188,35 @@ public static void pull() {
    * Shoots a fireball in direction the hero is facing.
    *
    * @param ac AmmunitionComponent of the hero, ammunition amount will be reduced by 1
+   * @param hero Entity to be used as hero for positioning
    */
-  private static void aimAndShoot(AmmunitionComponent ac) {
-    Entity hero = Game.hero().orElseThrow(MissingHeroException::new);
-    Direction viewDirection =
-        Direction.fromPositionCompDirection(EntityUtils.getViewDirection(hero));
-    Skill fireball =
-        new Skill(
-            new FireballSkill(
-                () ->
-                    hero.fetch(CollideComponent.class)
-                        .map(cc -> cc.center(hero))
-                        .map(p -> p.add(viewDirection.toPoint()))
-                        .orElseThrow(
-                            () -> MissingComponentException.build(hero, CollideComponent.class)),
-                FIREBALL_RANGE,
-                FIREBALL_SPEED,
-                FIREBALL_DMG),
-            1);
-    fireball.execute(hero);
+  private static void aimAndShoot(AmmunitionComponent ac, Entity hero) {
+    newFireballSkill(hero).execute(hero);
```

Comment:
das sieht wild aus

Reply:
Wegen 2x hero?

Die Sache ist das wir erst die Blickrichtung von dem Hero in der `newFireballSkill` raushohlen, um die Schießrichtung zu bestimmen